src/range_proof.rs

```diff
@@ -194,93 +252,203 @@ where
                     ));
                 } else {
-                    bit_vector_of_scalars(witness.openings[j].v - minimum_value, bit_length)?
+                    bit_vector_of_scalars(value - minimum_value, bit_length)?
                 }
             } else {
-                bit_vector_of_scalars(witness.openings[j].v, bit_length)?
+                bit_vector_of_scalars(values[j], bit_length)?
             };
             for bit_field in bit_vector.clone() {
                 a_li.push(bit_field);
                 a_ri.push(bit_field - Scalar::one());
             }
         }
 
-        // Compute A by multi-scalar multiplication
-        let rng = &mut thread_rng();
-        let mut alpha = Vec::with_capacity(extension_degree);
-        for k in 0..extension_degree {
-            alpha.push(if let Some(seed_nonce) = statement.seed_nonce {
-                nonce(&seed_nonce, "alpha", None, Some(k))?
-            } else {
-                // Zero is allowed by the protocol, but excluded by the implementation to be unambiguous
-                Scalar::random_not_zero(rng)
-            });
-        }
-        let mut ai_scalars = Vec::with_capacity(bit_length * aggregation_factor + extension_degree);
-        let mut ai_points = Vec::with_capacity(bit_length * aggregation_factor + extension_degree);
-        for k in 0..extension_degree {
-            ai_scalars.push(alpha[k]);
-            ai_points.push(g_base_vec[k].clone());
+        // Compute A by multi-scalar multiplication using the signer message
+        let a = P::vartime_multiscalar_mul(
+            a_li.iter().chain(a_ri.iter()),
+            statement
+                .generators
+                .gi_base_iter()
+                .chain(statement.generators.hi_base_iter()),
+        ) + signer_message.alpha_g;
+
+        // Start the transcript
+        let mut transcript = Transcript::new(transcript_label.as_bytes());
+        transcript.domain_separator(b"Bulletproofs+", b"Range Proof");
+        transcripts::transcript_initialize::<P>(
+            &mut transcript,
+            &statement.generators.h_base().compress(),
+            statement.generators.g_bases_compressed(),
+            bit_length,
+            extension_degree,
+            aggregation_factor,
+            &statement.commitments_compressed,
+            &statement.minimum_value_promises,
+        )?;
+
+        // Return the message and state
+        Ok((
+            HelperBitCommitmentMessage { a: a.compress() },
+            HelperBitCommitmentState {
+                transcript,
+                statement,
+                seed_nonce_helper,
+                a_li,
+                a_ri,
+                a,
+            },
+        ))
+    }
+
+    /// The signer receives the bit commitment message from the helper and produces the masked masks (yes, you read that
+    /// correctly)
+    #[allow(clippy::needless_pass_by_value)] // this ensures we consume the helper message
+    fn signer_mask_masks(
```

Comment:
As far as I read the comments and have been following the code, this should run on the signer device?
If that's the case this wont work at all. All of the data inside of the state is too large to run on a ledger. 
We have +- 26kb to work with, h_base + g_base is almost that alone. 

The signer can only do like a simple single math commitment, hashes etc. Transcripts I doubt it can.